test/token/ERC20/extensions/ERC20Wrapper.test.js

```diff
@@ -37,175 +40,169 @@ contract('ERC20Wrapper', function (accounts) {
+      await expect(tx)
+        .to.emit(this.underlying, 'Transfer')
+        .withArgs(this.token.target, this.initialHolder.address, initialSupply)
+        .to.emit(this.token, 'Transfer')
+        .withArgs(this.initialHolder.address, ethers.ZeroAddress, initialSupply);
+
+      await expect(tx).to.changeTokenBalances(
+        this.underlying,
+        [this.token, this.initialHolder],
+        [-initialSupply, initialSupply],
+      );
+      await expect(tx).to.changeTokenBalance(this.token, this.initialHolder, -initialSupply);
     });
 
     it('to other account', async function () {
-      const { tx } = await this.token.withdrawTo(receiver, initialSupply, { from: initialHolder });
-      await expectEvent.inTransaction(tx, this.underlying, 'Transfer', {
-        from: this.token.address,
-        to: receiver,
-        value: initialSupply,
-      });
-      await expectEvent.inTransaction(tx, this.token, 'Transfer', {
-        from: initialHolder,
-        to: ZERO_ADDRESS,
-        value: initialSupply,
-      });
+      const tx = await this.token.connect(this.initialHolder).withdrawTo(this.recipient, initialSupply);
+      await expect(tx)
+        .to.emit(this.underlying, 'Transfer')
+        .withArgs(this.token.target, this.recipient.address, initialSupply)
+        .to.emit(this.token, 'Transfer')
+        .withArgs(this.initialHolder.address, ethers.ZeroAddress, initialSupply);
+
+      await expect(tx).to.changeTokenBalances(
+        this.underlying,
+        [this.token, this.initialHolder, this.recipient],
+        [-initialSupply, 0, initialSupply],
+      );
+      await expect(tx).to.changeTokenBalance(this.token, this.initialHolder, -initialSupply);
     });
 
     it('reverts withdrawing to the wrapper contract', async function () {
-      expectRevertCustomError(
-        this.token.withdrawTo(this.token.address, initialSupply, { from: initialHolder }),
-        'ERC20InvalidReceiver',
-        [this.token.address],
-      );
+      await expect(this.token.connect(this.initialHolder).withdrawTo(this.token, initialSupply))
+        .to.be.revertedWithCustomError(this.token, 'ERC20InvalidReceiver')
+        .withArgs(this.token.target);
     });
   });
 
   describe('recover', function () {
     it('nothing to recover', async function () {
-      await this.underlying.approve(this.token.address, initialSupply, { from: initialHolder });
-      await this.token.depositFor(initialHolder, initialSupply, { from: initialHolder });
-
-      const { tx } = await this.token.$_recover(receiver);
-      await expectEvent.inTransaction(tx, this.token, 'Transfer', {
-        from: ZERO_ADDRESS,
-        to: receiver,
-        value: '0',
-      });
+      await this.underlying.connect(this.initialHolder).approve(this.token, initialSupply);
+      await this.token.connect(this.initialHolder).depositFor(this.initialHolder, initialSupply);
+
+      const tx = this.token.$_recover(this.recipient);
```

Comment:
Sometimes we await these `const tx =` or `this.tx =`, and sometimes we don't. I think we should be consistent.

If its works without the await, I'd use that everywhere.

Reply:
I tested remove then and they don't aways work

if the event is tested first it works because the 'await expect(tx) triggers the execution. 

But if the balance is tested with no other tests prior it will fail because the execution is only triggered by the await